### DISCUSSION TITLE
Can not display a MSI UI with a managed boostrapper and bundle using wix v4

### DISCUSSION THREAD

**Discussion by neech** [wixtoolset/issues#7131] [Questions]
I want to show the UI of a MSI included in a bundle, while searching on the wix v4 documentation I saw that the « DisplayInternalUI » attribute has been transferred to the balextension but I can not it except on the wix4 archive on git

**Comment by neech** [comment]
I investigated a bit and i noticed that balextension doesn't exist in wixv4 and had been replaced by Bal. 
Bal contains a displayUICondition attribute, but it works only when using <bal:WixStandardBootstrapperApplication /> and not with a custom WPF boostrapper UI using <bal:WixManagedBootstrapperApplicationHost  />

**Reply by mjanulaitis** [reply]
@neech I don't see search controls anywhere.  How did you search the docs?  I have the following in my bootstrapper but do not see the option.  Where should I be adding this?

```
<BootstrapperApplication>
	<bal:WixStandardBootstrapperApplication
		SuppressOptionsUI="yes"
		LicenseUrl=""
		LogoFile="..\Resources\cb.bmp"
		Theme="hyperlinkLicense" />
</BootstrapperApplication>
```

**Comment by rseanhall** [comment]
As requested by many, many people in v3, the BA controls when the internal MSI UI is shown. This is done in the `OnPlanMsiPackage` message.

wixstdba supports this functionality with the `bal:DisplayInternalUICondition` attribute. If it is missing from the docs, then please file a bug.

**Reply by mjanulaitis** [reply]
@rseanhall I am a little confused with the documentation because I do not see a search function.  I found the BootstrapperApplication bal:DisplayInternalUICondition but I don't see any docs on the condition.  I tried setting it to true but that did not work.  I am at a lose on how to set it.  I always want my UI to show.  Does that make sense?

**Reply by mjanulaitis** [reply]
I tried setting the condition to:

`<BootstrapperApplication bal:DisplayInternalUICondition="BURN_MSI_PROPERTY_INSTALL OR BURN_MSI_PROPERTY_MODIFY or BURN_MSI_PROPERTY_REPAIR">`

but that did not work.  Any idea's how to turn the MSI UI on?

**Reply by mjanulaitis** [reply]
I finally figured out how to get the internal UI to work using the WixInternalUIBootstrapperApplication however, the splash screen flickers then UAC prompts but the prompt is in the background.  If I double-click the exe from Explorer I see a quick flicker then I have to look at the task bar so see the UAC prompt.  It does not take focus.

**Reply by rseanhall** [reply]
`bal:DisplayInternalUICondition` is for `MsiPackage` elements. The condition engine does not have special names like `BURN_MSI_PROPERTY_INSTALL`, it would interpret that as a Variable name. WixInternalUIBootstrapperApplication is a completely separate BA from WixStandardBootstrapperApplication.
